From 52186b585d8834085c14fc088f1a36cd6184365c Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Tue, 12 Oct 2021 09:59:54 +0100
Subject: [PATCH] ARM: dts: Make vcio a child of the firmware node

In order to resolve a potential startup order bug, the vcio driver has
been rewritten as a platform driver that depends on a DT node for
its instantiation and to locate the firmware driver.

Add that DT node.

See: https://github.com/raspberrypi/linux/issues/4620

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm270x-rpi.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/arch/arm/boot/dts/bcm270x-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm270x-rpi.dtsi
@@ -152,3 +152,9 @@
 		status = "disabled";
 	};
 };
+
+&firmware {
+	vcio: vcio {
+		compatible = "raspberrypi,vcio";
+	};
+};
